services/data-migration/pipeline/transform.py

```diff
@@ -21,114 +25,90 @@ def create_organisation(row: pd.Series, current_timestamp: str) -> Organisation:
-) -> Endpoints:
-    return Endpoints(
+    endpoint_data: Dict[str, str], organisation_id: UUID, current_timestamp: str
+) -> Endpoint:
+    payload_type = (
+        None
+        if endpoint_data.get("transport") == "telno"
+        else endpoint_data.get("interaction")
+    )
+    format = (
+        None
+        if endpoint_data.get("transport") == "telno"
+        else endpoint_data.get("format")
+    )
+
+    return Endpoint(
         id=uuid4(),
-        identifier_oldDoS_id=endpoint_data.get("endpointid"),
+        identifier_oldDoS_id=endpoint_data.get("id"),
         status="active",
         connectionType=endpoint_data.get("transport"),
         name=None,
         description=endpoint_data.get("businessscenario"),
-        payloadType=""
-        if endpoint_data.get("transport") == "telno"
-        else endpoint_data.get("interaction"),
+        payloadType=payload_type,
         address=endpoint_data.get("address"),
         managedByOrganisation=organisation_id,
         service=None,
         order=endpoint_data.get("endpointorder"),
         isCompressionEnabled=endpoint_data.get("iscompressionenabled") == "compressed",
-        format=""
-        if endpoint_data.get("transport") == "telno"
-        else endpoint_data.get("format"),
+        format=format,
         createdBy="ROBOT",
         createdDateTime=current_timestamp,
         modifiedBy="ROBOT",
         modifiedDateTime=current_timestamp,
     )
 
 
-def convert_UUID_to_string(gp_dicts: dict) -> Dict[str, Any]:
-    for gp_dict in gp_dicts:
-        gp_dict["organisation"]["id"] = str(gp_dict["organisation"]["id"])
-        if gp_dict["endpoints"]:
-            for endpoint in gp_dict["endpoints"]:
-                endpoint["id"] = str(endpoint["id"])
-                endpoint["managedByOrganisation"] = str(
-                    endpoint["managedByOrganisation"]
-                )
-
-    return gp_dicts
-
-
-def create_GP_practices_dict(
-    df: pd.DataFrame, current_timestamp: str
-) -> List[Dict[str, Any]]:
+def transform_gp_practices(
+    df: pd.DataFrame, current_timestamp: datetime
+) -> pd.DataFrame:
     if df.empty:
-        logging.error("No data found")
-        return []
+        err_msg = "No data found in the input DataFrame"
+        raise ValueError(err_msg)
 
     gp_practices = []
     for _, row in df.iterrows():
         organisation = create_organisation(row, current_timestamp)
+
         endpoints_data = row.get("endpoints", [])
-        if endpoints_data is None:
-            logging.error("No endpoints found for the organisation")
-            gp_practices.append(
-                {"organisation": organisation.model_dump(), "endpoints": None}
-            )
-            continue
-
-        endpoints = [
-            create_endpoint(ed, organisation.id, current_timestamp)
-            for ed in endpoints_data
+        if endpoints_data is None or len(endpoints_data) == 0:
```

Comment:
when running I get an error here:
`TypeError: object of type 'float' has no len()`
`endpoints_data = nan`

```suggestion
               if endpoints_data is None or pd.isna(endpoints_data) or len(endpoints_data) == 0:
```